src/Components/RecommendationsPage/RecommendationsTable.tsx

```diff
@@ -28,9 +39,38 @@ const RecommendationsTable = ({ page, perPage, setTotal }) => {
   const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
   const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(null);
   const [recommendations, setRecommendations] = React.useState<Recommendation[]>([]);
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  const [currentRule, setCurrentRule] = useState<Recommendation | null>(null);
+  const [currentResolutions, setCurrentResolutions] = useState<Resolution | null>(null);
+
+  const ruleName = currentRule?.id.replace('leapp_report|', '') || '';
 
   const { getRegistry } = useContext(RegistryContext);
 
+  const getResolutionsInfo = async (rule: Recommendation) => {
+    setCurrentResolutions(null);
+    const response = await remediationsResolutions(rule.id);
+    if (isError(response)) {
+      const store = getRegistry().getStore();
+      displayErrorNotification(store, response.message);
+      return;
+    }
+    setCurrentResolutions(response[`advisor:${rule.id}`]?.resolutions[0]);
+  };
+
+  const createRemediation = async (selectedSystems) => {
+    if (!currentRule) return;
```

Comment:
Shouldn't here be some error notification if for some reason the rule is empty? Shouldn't probably happen :thinking:

Reply:
There should be no possible way to hit this :shrug: But I will add some error notif for sanity.